pkg/configs/api/api.go

```diff
@@ -164,27 +186,30 @@ func (a *API) validateAlertmanagerConfig(w http.ResponseWriter, r *http.Request)
 	})
 }
 
-func validateAlertmanagerConfig(cfg string) error {
+func validateAlertmanagerConfig(cfg string, noCfg NotificationsConfig) error {
 	amCfg, err := amconfig.Load(cfg)
 	if err != nil {
 		return err
 	}
 
 	for _, recv := range amCfg.Receivers {
-		if len(recv.EmailConfigs) != 0 {
-			return fmt.Errorf("email notifications are not supported in Cortex yet")
+		if noCfg.DisableEmail && len(recv.EmailConfigs) > 0 {
+			return fmt.Errorf("email notifications are disabled")
```

Comment:
If you define a global var for this err and the next one, then you could reference that error in tests (which makes tests more robusts).